extensions/amp-youtube/1.0/component.js

```diff
@@ -115,21 +114,52 @@ export function Youtube({
 
   src = addParamsToUrl(src, params);
 
+  // Passed through to imperative handle below.
+  const infoRef = useRef(
+    dict({
+      'currentTime': 0,
+      'duration': NaN,
+    })
+  );
+
+  useImperativeHandle(
+    ref,
+    () => {
+      const handle = {};
+      for (const key in infoRef.current) {
+        const get = () => infoRef.current[key];
+        Object.defineProperty(handle, key, {get});
```

Comment:
Nit: This kind of dynamic behavior makes it difficult to understand what the API is. Please document, or change to a declarative syntax.